src/components/shared/thread/ActivityGraphFills.js

```diff
@@ -271,13 +282,29 @@ export class ActivityGraphFillComputer {
         : greyCategoryIndex;
 
     let cpuBeforeSample = null;
-    if (enableCPUUsage && threadCPUDelta && threadCPUDelta[lastIdx] !== null) {
-      const cpuDelta = threadCPUDelta[lastIdx];
-      const intervalDistribution =
-        lastIdx === 0
-          ? 1
-          : (samples.time[lastIdx] - samples.time[lastIdx - 1]) / interval;
-      cpuBeforeSample = cpuDelta / intervalDistribution;
+    let cpuAfterSample = null;
+    if (enableCPUUsage && threadCPUDelta) {
+      const cpuDeltaBefore = ensureExists(threadCPUDelta[lastIdx]);
+      const intervalDistribution = (sampleTime - prevSampleTime) / interval;
+      cpuBeforeSample = cpuDeltaBefore / intervalDistribution;
+
+      const nextIdxInFullThread = sampleIndexOffset + lastIdx + 1;
+      if (nextIdxInFullThread < fullThread.samples.length) {
```

Comment:
This part became a bit more complicated but it was needed :/

Reply:
Yeah, I was trying to use the fullThread as my initial implementation but that didn't work out as expected. I had to change it to this implementation instead.